### PR TITLE
fix arch key + remove small yamls and keep the final yaml + update README

### PR DESCRIPTION
**Pull request by shay-berman**
**1. README update**
1.1. put master branch in the curl operations - make it ready for GA.
1.2. remove IBMBlockCSI yaml from the readme, and instead asking the user to create it, we just move it as a file that the user curl and download from github. the same as the operator yaml.
1.3. Use -l csi to get the pods

**2. Keep one yaml of the operator**
    Suggesting to keep only 1 yaml instead of manually generate it with hacky script - reduce code duplication and avoid human mistake for update one yaml and forget to run the hacky script to generate the final yaml. So i believe its better for simplisyty to keep the original final yaml (just git mv it to the top of the deploy directory)
- deploy/csi_driver.yaml
- deploy/operator.yaml
- deploy/role.yaml
- deploy/role_binding.yaml
- deploy/service_account.yaml
+ git mv /deploy/install/generated/ibm-block-csi-operator.yaml to  /deploy/ibm-block-csi-operator.yaml

**3. fix kubernetes.io/arch**  
instead of beta.kubernetes.io/arch which is deprecated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Please see [`Prerequisites for Driver Installation`](https://github.com/IBM/ibm-
 ### Install the operator
 1. Download the manifest from GitHub.
 ```bash
-curl https://raw.githubusercontent.com/IBM/ibm-block-csi-operator/develop/deploy/installer/generated/ibm-block-csi-operator.yaml > ibm-block-csi-operator.yaml
+curl https://raw.githubusercontent.com/IBM/ibm-block-csi-operator/master/deploy/ibm-block-csi-operator.yaml > ibm-block-csi-operator.yaml
```
Comment (27149chen) [inline README.md]: Why there are two PRs but update same thing?

File: README.md [hunk 2]
```diff
@@ -38,40 +38,36 @@ ibm-block-csi-operator-5bb7996b86-xntss 2/2     Running   0          10m
...
-    tag: "1.0.0"
-  node:
-    repository: ibmcom/ibm-block-csi-driver-node
-    tag: "1.0.0"
+
+
+1. Download the manifest from GitHub.
+```bash
+curl https://raw.githubusercontent.com/IBM/ibm-block-csi-operator/master/deploy/ibm-block-csi-operator-customer-resource.yaml > ibm-block-csi-operator-customer-resource.yaml
 ```
```
Comment (27149chen) [inline README.md]: update it

File: deploy/csi_driver.yaml [hunk 3]
```diff
@@ -1,7 +0,0 @@
-apiVersion: storage.k8s.io/v1beta1
-kind: CSIDriver
-metadata:
-  name: block.csi.ibm.com
-spec:
-  attachRequired: true
-  podInfoOnMount: false
```
Comment (27149chen) [inline deploy/csi_driver.yaml]: why to delete this file?

File: deploy/ibm-block-csi-operator-customer-resource.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: ibm-block-csi
+  namespace: kube-system
+spec:
+  controller:
+    repository: ibmcom/ibm-block-csi-driver-controller
+    tag: "1.0.0"
+  node:
+    repository: ibmcom/ibm-block-csi-driver-node
+    tag: "1.0.0"
```
Comment (27149chen) [inline deploy/ibm-block-csi-operator-customer-resource.yaml]: please don't add it

File: deploy/ibm-block-csi-operator.yaml [hunk 5]
```diff
@@ -1,5 +1,3 @@
-# Code generated by update-copyright.sh. DO NOT EDIT.
-
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
```
Comment (27149chen) [inline deploy/ibm-block-csi-operator.yaml]: please don't move this file

File: deploy/operator.yaml [hunk 6]
```diff
@@ -1,84 +0,0 @@
-apiVersion: apps/v1
```
Comment (27149chen) [inline deploy/operator.yaml]: please don't delete it.

File: deploy/role.yaml [hunk 7]
```diff
@@ -1,152 +0,0 @@
-apiVersion: rbac.authorization.k8s.io/v1
```
Comment (27149chen) [inline deploy/role.yaml]: please don't delete it.

File: deploy/role_binding.yaml [hunk 8]
```diff
@@ -1,15 +0,0 @@
-kind: ClusterRoleBinding
```
Comment (27149chen) [inline deploy/role_binding.yaml]: please don't delete it.

File: deploy/service_account.yaml [hunk 9]
```diff
@@ -1,8 +0,0 @@
-apiVersion: v1
```
Comment (27149chen) [inline deploy/service_account.yaml]: please don't delete it.

Comment (shay-berman) [reply deploy/service_account.yaml]: yes, I opened new PR to avoid the deletion. please ignore this PR, delete it now

**Comment by shay-berman** [comment]
opened a new PR https://github.com/IBM/ibm-block-csi-operator/pull/12  to make it cleaner (after the discussion with Lou)

**Pull request closed without merging**